tests/test_helpers.py

```diff
@@ -184,3 +192,101 @@ def test_generate_citation_with_all_fields():
 def test_parse_xml_records_returns_record_iterator():
     records = parse_xml_records("tests/fixtures/datacite/datacite_records.xml")
     assert len(list(records)) == 38
+
+
+def test_parse_date_from_string_success():
+    for date in [
+        "1930",
+        "1930-12",
+        "1930",
+        "1930-12",
+        "1930-12-31",
+        "1930-12-31",
+        "19301231",
+        "19301231",
+        "12/31/1930",
+        "12/31/1930",
+        "1930/12/31",
+        "1930T12",
+        "1930T12Z",
+        "1930T12:34",
+        "1930T12:34Z",
+        "1930T12:34:56",
+        "1930T12:34:56Z",
+        "1930T12:34:56.000001",
+        "1930T12:34:56.000001Z",
+        "1930-12T12",
+        "1930-12T12Z",
+        "1930-12T12:34",
+        "1930-12T12:34Z",
+        "1930-12T12:34:56",
+        "1930-12T12:34:56Z",
+        "1930-12T12:34:56.000001",
+        "1930-12T12:34:56.000001Z",
+        "1930-12-31T12",
+        "1930-12-31T12Z",
+        "1930-12-31T12:34",
+        "1930-12-31T12:34Z",
+        "1930-12-31T12:34:56",
+        "1930-12-31T12:34:56Z",
+        "1930-12-31T12:34:56.000001",
+        "1930-12-31T12:34:56.000001Z",
+        "1930T12",
+        "1930T12Z",
+        "1930T12:34",
+        "1930T12:34Z",
+        "1930T12:34:56",
+        "1930T12:34:56Z",
+        "1930T12:34:56.000001",
+        "1930T12:34:56.000001Z",
+        "1930-12T12",
+        "1930-12T12Z",
+        "1930-12T12:34",
+        "1930-12T12:34Z",
+        "1930-12T12:34:56",
+        "1930-12T12:34:56Z",
+        "1930-12T12:34:56.000001",
+        "1930-12T12:34:56.000001Z",
+        "1930-12-31T12",
+        "1930-12-31T12Z",
+        "1930-12-31T12:34",
+        "1930-12-31T12:34Z",
+        "1930-12-31T12:34:56",
+        "1930-12-31T12:34:56Z",
+        "1930-12-31T12:34:56.000001",
+        "1930-12-31T12:34:56.000001Z",
+    ]:
+        assert type(parse_date_from_string(date)) == datetime
+
+
+def test_parse_date_from_string_invalid_date_returns_false():
+    assert not parse_date_from_string("circa 1930s")
+
+
+def test_validate_date_success():
+    assert validate_date("1930 ", "1234") is True
+
+
+def test_validate_date_invalid_date_logs_error(caplog):
```

Comment:
This should also confirm it returns `False`. Same for date range validation tests below.